mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -55,6 +55,10 @@ class DecisionTreeClassifier @Since("1.4.0") (
 
   // Override parameter setters from parent trait for Java API compatibility.
 
+  /** @group setParam */
+  @Since("3.0.0")
+  def setLeafCol(value: String): this.type = set(leafCol, value)
```

Comment:
I think we can do some refactoring here, in order to dedup this. Can we add it to a trait?

Reply:
This way seems like mllib's convention that not add setter into the `xxxParam`-like trait,  like `setVarianceCol` in `DecisionTreeRegressionModel`

Reply:
Yes, agree, it shouldn't be in the model classes. So `DecisionTreeClassifierParams` doesn't help. Hm... per the discussion below, I agree it's extra weight to refactor the common elements into a superclass of two decision tree classifiers, but it might well be worth it. It looks like it would save a few hundred lines of duplicated code? that would mitigate the concern about the large change here. I'm lightly in favor of going that way. I wouldn't do it for 10 lines of code or something.